components/brave_wallet_ui/page/screens/composer_ui/token_list_item/token_list_item.tsx

```diff
@@ -67,151 +74,212 @@ export const TokenListItem = React.forwardRef<HTMLDivElement, Props>(
     } = props
 
     // Queries
-    const { data: tokensNetwork } = useGetNetworkQuery(token ?? skipToken)
+    const { data: tokensNetwork } = useGetNetworkQuery(token.asset ?? skipToken)
     const { data: defaultFiatCurrency = 'usd' } =
       useGetDefaultFiatCurrencyQuery()
 
     const fiatBalance = React.useMemo(() => {
-      return balance && spotPrice
-        ? new Amount(balance).divideByDecimals(token.decimals).times(spotPrice)
+      return token.assetBalance && spotPrice
+        ? new Amount(token.assetBalance)
+            .divideByDecimals(token.asset.decimals)
+            .times(spotPrice.price)
         : Amount.empty()
-    }, [spotPrice, balance, token.decimals])
+    }, [spotPrice, token])
 
     const formattedFiatBalance = fiatBalance.formatAsFiat(defaultFiatCurrency)
 
-    const networkDescription = React.useMemo(() => {
-      if (tokensNetwork) {
-        return token.symbol !== ''
-          ? getLocale('braveWalletPortfolioAssetNetworkDescription')
-              .replace('$1', token.symbol)
-              .replace('$2', tokensNetwork.chainName ?? '')
-          : tokensNetwork.chainName
-      }
-      return token.symbol
-    }, [tokensNetwork, token.symbol])
-
     const tokenDisplayName = React.useMemo(() => {
-      if (token.isNft) {
-        const id = token.tokenId
-          ? `#${new Amount(token.tokenId).toNumber()}`
+      if (token.asset.isNft) {
+        const id = token.asset.tokenId
+          ? `#${new Amount(token.asset.tokenId).toNumber()}`
           : ''
-        return `${token.name} ${id}`
+        return `${token.asset.name} ${id}`
       }
       if (disabledText) {
-        return token.name.length > 12
-          ? `${token.name.substring(0, 10)}...`
-          : token.name
+        return token.asset.name.length > 12
+          ? `${token.asset.name.substring(0, 10)}...`
+          : token.asset.name
       }
-      return token.name
+      return token.asset.name
     }, [token, disabledText])
 
+    const tokenHasBalance = new Amount(token.assetBalance).gt(0)
+
+    const isPriceDown = spotPrice
+      ? Number(spotPrice.assetTimeframeChange) < 0
+      : false
+
     return (
-      <ButtonWrapper
-        width='100%'
+      <Column
+        fullWidth={true}
         ref={forwardedRef}
       >
-        <Button
-          onClick={onClick}
-          disabled={!!disabledText}
+        {label === 'owned' && (
+          <Row
+            justifyContent='space-between'
+            padding='10px 16px'
+          >
+            <Text
+              textSize='12px'
+              isBold={false}
+              textColor='secondary'
+            >
+              {getLocale('braveWalletOwned')}
+            </Text>
+            <Text
+              textSize='12px'
+              isBold={false}
+              textColor='secondary'
+            >
+              {getLocale('braveWalletAmount24H')}
+            </Text>
+          </Row>
+        )}
+        {label === 'not-owned' && (
```

Comment:
perf: combine ternary to avoid checking `label` twice

Reply:
Changed!